testing/087_call_graph.c

```diff
@@ -0,0 +1,36 @@
+// objective: root and truncated nodes
+// check: 087__call__graph_8c_ab50e9ccc8158d9b5cab04dba22244fbd_cgraph.dot
+// config: DOT_GRAPH_MAX_NODES=8
+// config: GENERATE_HTML=YES
+// config: HAVE_DOT=YES
+// config: CALL_GRAPH=YES
+// config: CALLER_GRAPH=YES
+// config: DOT_CLEANUP=NO
+// config: EXTRACT_ALL=YES
+// config: HTML_OUTPUT=out
```

Comment:
Will be automatically set when run in html mode.

Reply:
tests run automatically on command **make test** and it is right.
How to define my test html mode and run it automatically?

Reply:
The tests are not intended for automatic runs outside in xml mode the other modes are created for internal development testing (I found quite a few problems through it). The other modes don't test against the given files (here: "check: 087__call__graph_8c_ab50e9ccc8158d9b5cab04dba22244fbd_cgraph.dot") but do only some sanity checks on the format or convert to that format so it is known whether or not the format builds and the results can be inspected (e.g. rtf, pdf, docbook, chm). Interesting in that case is also the option `--keep`. 

I just ran your test case with (the old code) as `make tests TEST_FLAGS="--xml --xhtml --id=87 --keep"` but I got some messages:
```
Syntax of value for attribute id of map is not valid
...
Syntax of value for attribute id of map is not valid
Document test_output_087/html/087__callgraph_8c.xhtml does not validate
``` 
looks like the original already has some problem. The lines are like:
```
<map name="087__callgraph_8c_a1c4791d28548770b6e1bf28b18c95e96_cgraph" id="087__callgraph_8c_a1c4791d28548770b6e1bf28b18c95e96_cgraph">
```
I think I know the problem: an id cannot start with a number.  I also thought I fixed it (but apparently not or not everywhere, vaguely remember I had a discussion regarding xml / docbook about this, but I see in the xml also id starting with a number but these are internal names). This is something I have to look into.

Another problem might be that the `dot` program is not installed on travis and appveyor (I'm not sure).

Reply:
I've replaced testing/087_call_graph.c with examples/call_graph.c

Reply:
@albert-github

Reply:
What is the question?